lua/apple-music/init.lua

```diff
@@ -180,6 +186,34 @@ M.disable_shuffle = function()
 	end
 end
 
+---Change the favorited state of the current music.
+---It also handles Mac OS versions below 14 (Sonoma) where it used to be called "loved".
+---@param state boolean: The state of favorited to be set for current track.
+---@usage require('apple-music').set_current_track_favorited(true)
+---@usage require('apple-music').set_current_track_favorited(false)
+M.set_current_track_favorited = function(state)
```

Comment:
I think this functionality would be better if we split it into three:

1. `favorite_current()`
2. `unfavorite_current()`
3. `toggle_favorite_track()` (If we can get "is_favorited")

I think most people would use a keybinding and not want to worry about parameters. 

Don't have to worry about `toggle_favorite_track`, that can be a separate PR.

Reply:
Yeah, probably it'd be simpler to map this way. The favorited property can be fetched from the current track so the toggle should be feasible. 👍🏼

Reply:
I still kind of left this function there, but as a local one so we have less duplication and it can be also reused later on. But now we have two functions for this.